snapshot/kaniko_snapshot_unix_test.go

```diff
@@ -0,0 +1,157 @@
+// +build linux darwin
+
+package snapshot_test
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/sclevine/spec"
+
+	"github.com/buildpacks/lifecycle/snapshot"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+func TestKanikoSnapshotter(t *testing.T) {
+	spec.Run(t, "Test Image", testKanikoSnapshotter)
+}
+
+func testKanikoSnapshotter(t *testing.T, when spec.G, it spec.S) {
+	var (
+		snapshotter *snapshot.KanikoSnapshotter
+		tmpDir      string
+	)
+
+	it.Before(func() {
+		// Using the default tmp dir causes kaniko to go haywire for some reason
+		cwd, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("Error: %s\n", err)
+		}
+		tmpDir, err = ioutil.TempDir(filepath.Join(cwd, "..", "tmp"), "kaniko")
+		if err != nil {
+			t.Fatalf("Error: %s\n", err)
+		}
+
+		createTestFile(t, filepath.Join(tmpDir, "cnb", "privatefile"))
+		createTestFile(t, filepath.Join(tmpDir, "layers", "privatefile"))
+		createTestFile(t, filepath.Join(tmpDir, "file-to-change"))
+		createTestFile(t, filepath.Join(tmpDir, "file-not-to-change"))
+		createTestFile(t, filepath.Join(tmpDir, "file-to-delete"))
+		createTestFile(t, filepath.Join(tmpDir, "bin", "file-not-to-change"))
+
+		snapshotter = &snapshot.KanikoSnapshotter{
+			RootDir:                    tmpDir,
+			DetectFilesystemIgnoreList: false,
+		}
+		snapshotter.IgnoredPaths = []string{filepath.Join(snapshotter.RootDir, "dir-with-ignored-files")}
+	})
+
+	it.After(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	when("files are added and modified", func() {
+		var (
+			snapshotFile string
+		)
+
+		it.Before(func() {
+			h.AssertNil(t, snapshotter.Init())
+
+			os.Remove(filepath.Join(snapshotter.RootDir, "file-to-delete"))
+			createTestFileWithContent(t, filepath.Join(snapshotter.RootDir, "file-to-change"), "hola\n")
+			createTestFile(t, filepath.Join(snapshotter.RootDir, "my-space", "newfile-in-dir"))
```

Comment:
If you change this to `createTestFileWithContent` that could make the assertion more understandable (don't have to check the definition of `createTestFile` to know why we're expecting `hello`).